crates/types/src/storage.rs

```diff
@@ -226,3 +191,47 @@ macro_rules! flexbuffers_storage_encode_decode {
         }
     };
 }
+
+/// Utility method to encode a [`Serialize`] type as flexbuffers using serde.
+pub fn encode_as_flexbuffers<T: Serialize, B: BufMut>(
+    value: T,
+    buf: &mut B,
+) -> Result<(), flexbuffers::SerializationError> {
+    let vec = flexbuffers::to_vec(value)?;
+    // write the length
+    buf.put_u32_le(u32::try_from(vec.len()).map_err(|_| {
+        flexbuffers::SerializationError::custom("only support serializing types of size <= 4GB")
+    })?);
+    buf.put(&vec[..]);
```

Comment:
Can we check that the buffer has enough space? Bytes have the habit of panicking if buffer is small.

Reply:
True. Will change it.